src/connection/host_lookup.c

```diff
@@ -0,0 +1,591 @@
+      if (hosts_conf_file_Load == LOAD_INIT || hosts_conf_file_Load == LOAD_FAIL)
+	{
+//err_set : load fail
+	  return NULL;
+	}
+    }
+
+  addr_trans = (struct sockaddr_in *) saddr;
+
+  assert (((hostname != NULL) && (saddr == NULL)) || ((hostname == NULL) && (saddr != NULL)));
+
+  if (lookup_case == IPADDR_TO_HOSTNAME)
+    {
+      if (inet_ntop (AF_INET, &addr_trans->sin_addr, addr_trans_ch_buf, sizeof (addr_trans_ch_buf)) == NULL)
+	{
+//err_msg : convert binary to text fail
+	  return NULL;
+	}
+
+    }
+
+  /*Look up in the user_host_Map */
+  if ((lookup_case == HOSTNAME_TO_IPADDR) && (user_host_Map.find (hostname) != user_host_Map.end ()))
+    {
+      strcpy (ipaddr_buf, user_host_Map.find (hostname)->second.c_str ());
+      strcpy (hostname_buf, hostname);
+    }
+  else if ((lookup_case == IPADDR_TO_HOSTNAME) && (user_host_Map.find (addr_trans_ch_buf) != user_host_Map.end ()))
+    {
+      strcpy (hostname_buf, user_host_Map.find (addr_trans_ch_buf)->second.c_str ());
+      strcpy (ipaddr_buf, addr_trans_ch_buf);
+    }
+  else
+    {
+//err_msg : hosts.conf has no info
+      return NULL;
+    }
+
+  if (inet_pton (AF_INET, ipaddr_buf, addr_trans_bi_buf) < 1)
+    {
+//err_set : convert text to binary fail -> reverse
+      return NULL;
+    }
+
+  /*set the hostent struct for the return value */
+  if ((hp = hostent_init ()) == NULL)
+    {
+      return NULL;
+    }
+
+  strcpy (hp->h_name, hostname_buf);
+  strcpy (hp->h_aliases[0], hostname_buf);
+  memcpy (hp->h_addr_list[0], addr_trans_bi_buf, sizeof (hp->h_addr_list[0]));
+
+  return hp;
+}
+
+static int
+host_conf_load ()
+{
+  FILE *fp;
+  char file_line[HOSTNAME_BUF_SIZE + 1];
+  char line_buf[HOSTNAME_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token;
+  char *save_ptr_strtok;
+  /*delimiter */
+  char *delim = " \t\n";
+  char map_ipaddr[IPADDR_LEN];
+  char map_hostname[HOSTNAME_BUF_SIZE + 1];
+
+  /*True, when the string token has hostname, otherwise, string token has IP address */
+  bool hostent_flag;
+  HOSTENT_INSERT_TYPE hostent_insert_Type;
+
+  memset (file_line, 0, HOSTNAME_BUF_SIZE + 1);
+  memset (line_buf, 0, HOSTNAME_BUF_SIZE + 1);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, "hosts.conf");
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+//err_msg : file open fail
+//er_set_with_oserror (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_MOUNT_FAIL, 1, host_conf_file_full_path);
+//fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_IO_MOUNT_FAIL));
+      return LOAD_FAIL;
+    }
+
+  while (fgets (file_line, HOSTNAME_BUF_SIZE + 1, fp) != NULL)
+    {
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      hostent_flag = INSERT_IPADDR;
+
+      do
+	{
+	  if (*token == '\0' || *token == '#')
```

Comment:
**strtok_r** will return NULL if there are no more tokens. In this case, '***token**' will cause a coredump. Please check whether **token** is NULL first.

Reply:
```suggestion
	  if (*token ==NULL || *token == '#')
```

I fixed it like above.